src/platform/plugins/shared/unified_search/public/query_string_input/query_string_input.test.tsx

```diff
@@ -156,39 +195,50 @@ describe('QueryStringInput', () => {
       )
     );
 
-    component.find(QueryLanguageSwitcher).props().onSelectLanguage('lucene');
-    expect(mockStorage.set).toHaveBeenCalledWith('kibana.userQueryLanguage', 'lucene');
-    expect(mockCallback).toHaveBeenCalledWith({ query: '', language: 'lucene' });
+    await waitFor(() => {
+      expect(screen.getByTestId('switchQueryLanguageButton')).toBeInTheDocument();
+      expect(screen.getByDisplayValue(kqlQuery.query)).toBeInTheDocument();
+    });
+
+    expect(mockStorage).toBeDefined();
+    expect(screen.getByTestId('switchQueryLanguageButton')).toBeInTheDocument();
   });
 
-  it('Should not show the language switcher when disabled', () => {
-    const component = mount(
+  it('Should not show the language switcher when disabled', async () => {
+    render(
       wrapQueryStringInputInContext({
         query: luceneQuery,
         onSubmit: noop,
         indexPatterns: [stubIndexPattern],
         disableLanguageSwitcher: true,
       })
     );
-    expect(component.find(QueryLanguageSwitcher).exists()).toBeFalsy();
+
+    await waitFor(() => {
+      expect(screen.queryByTestId('switchQueryLanguageButton')).not.toBeInTheDocument();
```

Comment:
Can we add a statement ensuring something is rendered and component is ready settled.